<h1>Cálculos dos testes A/B</h1>

In [ ]:
# Imports para raiz quadrada, média, variância

# pip install scipy

import math
import statistics
import scipy.stats as st 

<h2>Dados Teste A</h2>

In [ ]:
# Tempo de resolução de cada user
tempos_resolucao_a = [73.44, 51.41, 29.79, 116.54, 22.2, 30.1, 87.9, 55.04, 35.78, 27.28,
                      58.27, 48.98, 38.08, 36.88, 70.17, 68.17, 47.38, 41.88, 27.59, 45.98,
                      42.95, 56.64, 57.75, 91.77, 38.03, 51.62, 114.94, 61.95, 88.05, 59.71,
                      107.52, 71.88, 23.48, 85.53, 22.43, 92.78, 25.27, 106.93, 78.02, 41.08,
                      29.19, 48.04, 18.38, 38.02, 36.44, 29.78, 31.98, 45.51, 28.75, 48.34,
                      36.59, 33.57, 16.34, 40.91]

# Variância amostral do tempo de resolução
variancia_a = statistics.variance(tempos_resolucao_a)

# Média do tempo de resolução em segundos
tempo_medio_a = statistics.mean(tempos_resolucao_a) 

# Quantidade total amostras
amostra_a = len(tempos_resolucao_a)

<h2>Dados Teste B</h2>

In [ ]:
# Tempo de resolução de cada user
tempos_resolucao_b = [27.7, 39.11, 47.55, 30.01, 12.73, 26.68, 117.26, 27.83, 35.76, 37.71,
                      43.88, 25.79, 60.66, 69.92, 56.19, 33.83, 45.09, 45.18, 24.43, 63.32,
                      88.99, 23.82, 49.35, 63.16, 70.2, 35.44, 47.02, 22.52, 72.41, 19.77,
                      36.41, 48.3, 91.62, 41.39, 47.95, 81.1, 20.3, 18.73, 66.64, 17.3,
                      8.24, 32.44, 36.86, 31.6, 34.85, 36.05, 24.28, 29.86, 53.04, 38.45,
                      43.78, 24.45, 51.3, 36.81]

# Variância amostral do tempo de resolução
variancia_b = statistics.variance(tempos_resolucao_b)

# Média do tempo de resolução em segundos
tempo_medio_b = statistics.mean(tempos_resolucao_b) 

# Quantidade total de amostras
amostra_b = len(tempos_resolucao_b)

42.87148148148148


<h2>Z Test</h2>

In [4]:
# XA e XB: São as médias de tempo.
numerador = (tempo_medio_a - tempo_medio_b)

# S²A e S²B: São as variâncias
# NA e NB: São o tamanho das amostras
denominador = (variancia_a/amostra_a)  +  (variancia_b/amostra_b)

# Raiz: math.sqrt()
denominador = math.sqrt(denominador)

z = (numerador / denominador)


<h2>Conclusão</h2>

In [ ]:
try:
    confianca = int(input("Qual o intervalo de confiança? (1-99)\n"))

    if 0 < confianca < 100:
        significancia_percentual = 100 - confianca
        significancia = significancia_percentual / 100.0        
        print(f"Intervalo de Confiança: {confianca}%")
        print(f"Nível de Significância: {significancia}")

        z_critico = st.norm.ppf(1 - significancia / 2)
        print(f"Z-Crítico aproximado: {z_critico:.4f}")

        print("\n\033[94mConclusão do Teste\033[0m")
        
        if z > 0:
            # tempo_medio_a > tempo_medio_b
            performance = "O Teste B performou melhor"
            vencedor = "o Teste B"
        elif z < 0:
            # tempo_medio_a < tempo_medio_b
            performance = "O Teste A performou melhor"
            vencedor = "o Teste A"
        else:
            performance = "Ambos os testes tiveram a mesma performance média"
            vencedor = "nenhum teste"
        
        if abs(z) > z_critico:
            # SIGNIFICATIVO
            print("\033[92m Resultado: A diferença entre os grupos é estatisticamente significativa.\033[0m")
            print(f"  Performance: {performance}.")
            print(f"  Justificativa: O Z-Score ({abs(z):.2f}) é maior que o Z-Crítico ({z_critico:.2f}).")
            print(f"  Conclusão: Podemos afirmar com {confianca}% de confiança que {vencedor} teve melhor performace.")
        
        else:
            # NÃO SIGNIFICATIVO
            print("\033[92m Resultado: A diferença entre os grupos não é estatisticamente significativa.\033[0m")
            print(f"  Performance: {performance}, porém a diferença não é significativa.")
            print(f"  Justificativa: O Z-Score ({abs(z):.2f}) é menor ou igual ao Z-Crítico ({z_critico:.2f}).")
            print(f"  Conclusão: Embora {vencedor} pareça ser mais rápido, não temos evidência estatística para provar que isso não foi por acaso.")
            
    else:
        print("Tente novamente! Insira um número entre 1 e 99.")
        
except ValueError:
    print("Tente novamente! Digite apenas números.")

<h1 style="text-align: center;">Relatório de Teste de Usabilidade (A/B)</h1>

<h2 style="color: #FF5733;">1. Introdução e Contexto</h2>
<hr>
<p>Este teste A/B foi proposto para validar se um novo design e fluxo (Teste B), focado em simplicidade e clareza, poderia melhorar as métricas de performance.</p>

<h2 style="color: #FF5733;">2. Hipótese</h2>
<hr>
<p>Nossa hipótese era que o <strong>Fluxo B</strong>, com menos telas, uma paleta de cores ajustada e botões reorganizados, resultaria em uma redução estatisticamente significativa no tempo médio de conclusão da tarefa e em uma menor percepção de dificuldade por parte dos usuários.</p>

<h2 style="color: #FF5733;">3. Metodologia: O que mudou (Fluxo B)</h2>
<hr>
<p>As principais mudanças implementadas no Fluxo B (comparado ao Fluxo A) incluem:</p>
<ul>
  <li>
    <strong>Redução no número de telas:</strong> O objetivo foi diminuir a insegurança e aumentar a confiança do usuário durante o processo.
  </li>
  <li>
    <strong>Ajuste na paleta de cores:</strong> O tom de laranja principal foi escurecido, buscando uma semelhança com o pôr do sol para maior conforto visual.
  </li>
  <li>
    <strong>Reorganização de botões:</strong> A ordem de alguns elementos foi alterada para criar um fluxo de navegação mais lógico e agradável.
  </li>
</ul>

<h2 style="color: #FF5733;">4. Coleta de Dados (Amostra)</h2>
<hr>
<p>O público-alvo dos testes incluiu pessoas de 16 a 40 anos. Obtivemos um total de 100 amostras, compostas por: 10 participantes ligados ao próprio negócio (engenheiros de alimentos e pessoas próximas), colaboradores da empresa PicPay e alunos profissionais selecionados do Instituto J&F.</p>

<h2 style="color: #FF5733;">5. Resultados Quantitativos (Métricas)</h2>
<hr>
<p>Para a análise, comparamos a média de tempo gasto em cada fluxo. Utilizamos um <strong>intervalo de confiança de 95%</strong>, o que significa que temos 95% de certeza de que a diferença observada é real e não obra do acaso.</p>
<p>Os resultados de tempo médio foram:</p>
<ul>
  <li><strong>Fluxo A (Controle):</strong> 52.09 segundos</li>
  <li><strong>Fluxo B (Variante):</strong> 42.87 segundos</li>
</ul>
<p>Considerando este intervalo de confiança, podemos afirmar que o <strong>Teste B teve uma performance superior</strong>, apresentando uma redução de aproximadamente 17,7% no tempo médio de conclusão.</p>

<h2 style="color: #FF5733;">6. Resultados Qualitativos (Observações)</h2>
<hr>
<p>Embora o Fluxo B tenha sido numericamente superior, observamos alguns pontos de atenção:</p>
<p>Alguns participantes demonstraram dificuldade considerável (os <i>outliers</i>). Isso ocorreu mesmo no Fluxo B. Usuários ficaram perdidos na tela inicial.</p>
<p>Por outro lado, uma parte dos usuários apresentou extrema facilidade, concluindo as tarefas propostas em poucos segundos. Isso demonstra que a amostra de testes foi bastante diversificada em termos de habilidade.</p>

<h2 style="color: #FF5733;">7. Conclusão e Próximos Passos</h2>
<hr>
<p><strong>Decisão:</strong> O Teste B é o vencedor. Recomendamos a implementação do Fluxo B para 100% dos usuários.</p>
<p><strong>Ações Futuras (Próximos Passos):</strong></p>
<ul>
  <li>Antes do lançamento final, faremos um ajuste nos itens citados para mitigar o problema.</li>
</ul>